source/D2Common/src/Drlg/DrlgPreset.cpp

```diff
@@ -224,35 +208,32 @@ void __fastcall DRLGPRESET_ParseDS1File(D2DrlgFileStrc* pDrlgFile, void* pMemPoo
 	}
 
 	pDrlgFile->pShadowLayer = pData;
-	pData = &pData[nArea];
+	SkipInt32s(pData, nArea);
 	if (pDrlgFile->unk0x00 > 0 && pDrlgFile->unk0x00 <= 2)
 	{
 		pDrlgFile->pSubstGroupTags = pData;
-		pData += nArea;
+		SkipInt32s(pData, nArea);
 	}
 
-	if (v72 > 1)
+	if (nVersion > 1)
 	{
-		nUnits = *pData;
-		++pData;
+		int nUnits = ReadInt32(pData);
 
 		for (int i = 0; i < nUnits; ++i)
 		{
-			nUnitType = *pData;
-			++pData;
-			nUnitId = *pData;
-			++pData;
+			int nUnitType = ReadInt32(pData);
+			int nUnitId = ReadInt32(pData);
 
-			nMode = 0;
+			int nMode = 0;
 
 			switch (nUnitType)
 			{
 			case UNIT_MONSTER:
 				nMode = MONMODE_NEUTRAL;
-				if (v72 > 4)
+				if (nVersion > 4)
 				{
-					nMonPresetRecords = 0;
-					pMonPresetTxtSection = DATATBLS_GetMonPresetTxtActSection(nAct, &nMonPresetRecords);
+					int nMonPresetRecords = 0;
+					D2MonPresetTxt* pMonPresetTxtSection = DATATBLS_GetMonPresetTxtActSection(nAct, &nMonPresetRecords);
 
 					if (pMonPresetTxtSection && nUnitId < nMonPresetRecords)
 					{
```

Comment:
@Lectem The bug mentioned in #10 is occuring here. The code has to look like this, i.e. the type has to be looked up before overwriting the id

`						const uint8_t nType = pMonPresetTxtSection[nUnitId].nType;

						nUnitId = pMonPresetTxtSection[nUnitId].wPlace;

						switch (nType)
`

Reply:
Nice catch, my unit tests didn't catch it due to the fact txt files data was not loaded, so it would just skip this part (both in the original game and using the d2moo function). This led to identical results for both functions, as the buggy code was never executed.
I added an assert to check this so that it at least users are warned when misused or assets are not correct.